src/emulator/storage/index.ts

```diff
@@ -42,13 +48,23 @@ export class StorageEmulator implements EmulatorInstance {
       getAdminCredentialValidator(),
       this._persistence
     );
+    this._adminStorageLayer = new StorageLayer(
+      args.projectId,
+      getAdminOnlyRulesValidator(),
+      getAdminCredentialValidator(),
```

Comment:
Huh. I don't know about making another `StorageLayer`. If the purpose is just to validate requests a different way, why do you need to separate all the stored objects? Is there any reason you can't do what you did for `AdminCredentialValidator`?